### ISSUE TITLE
Key storage directory is not in an appropriate config location

### ISSUE THREAD

**Issue by andydotxyz** [AletheiaWareLLC/BC#1]
On Linux I would have expected ~/.config/bc/keys, but instead it is in my home directory ~/bc/keys.
I don't know if it is configurable or not, but the default should be less intrusive.

**Comment by stuartmscott** [comment]
Yes it is configurable with the `KEY_DIRECTORY` env var (similarly the cache can be configured with `CACHE_DIRECTORY`, or the entire `bc` root can be configured with `ROOT_DIRECTORY`), but I agree the defaults could be better as the directory need not be visible to everyday users.

There are a two options here, so I'll need to investigate the best approach - suggestions appreciated.
- Use the `~/.config` directory on Linux as you suggested, and the equivalent on the other platforms. But I believe the only folders Android provides are `files` and `cache`, the latter is already used as `CACHE_DIRECTORY`.
- Use `~/.bc` so the root directory is hidden on all platforms.

Either way there will need to be some migration code to ensure the current folder still works for existing users.

**Comment by andydotxyz** [comment]
I would go platform specific. As far as I was aware hidden on windows is different to whether it starts with a “.”, so I would
Think it may still be visible (for some versions?). I think iOS and Android are always special as they have a sandbox, so I don’t think there is a single solution for all.

**Comment by stuartmscott** [comment]
Ah yes, good point - the '.' prefix doesn't work on all platforms.

Linux, Mac, and Windows should use platform-specific hidden directories.

I think `{app}/files/bc/` is okay for Android & iOS as you said these are sandboxes, and not typically browsable by end users.

**Comment by andydotxyz** [comment]
Sounds good.
